friends/src/components/PrivateRoute.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { Route, Redirect } from 'react-router-dom'
+
+function PrivateRoute(props) {
+	const {
+		component: Component,
+		...rest
+	} = props
+
+	return (
+		<Route {...rest} render={(props) => {
+			const token = localStorage.getItem('token')
+				return token
+					? <Component />
```

Comment:
inside the Component, you need to spread in the props from line 11.  These props contain the router props that allow us to navigate/use match for the url